pkg/tar/tar.go

```diff
@@ -106,36 +134,43 @@ func (t *stiTar) StreamDirAsTar(source, dest string, writer io.Writer) error {
 	if !info.IsDir() {
 		return fmt.Errorf("the source %q has to be directory, not a file", source)
 	}
-	fs := util.NewFileSystem()
-	tmpDir, err := ioutil.TempDir("", "s2i-")
-	if err != nil {
-		return err
-	}
-	defer os.RemoveAll(tmpDir)
-	if err = fs.Copy(source, tmpDir); err != nil {
-		return err
-	}
-	// Skip chmod if on windows OS
-	if runtime.GOOS != "windows" {
-		err = filepath.Walk(tmpDir, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-			if info.IsDir() {
-				return os.Chmod(path, 0777)
-			}
-			return os.Chmod(path, 0666)
-		})
+	destDir := source
+	if !modifyInplace {
```

Comment:
well avoiding the copy would be the default behavior i'd expect, so the question is under what conditions is it necessary to make the copy?